test/e2e/drain_tool_test.go

```diff
@@ -0,0 +1,283 @@
+
+	It("drain cluster using binary, should succeed", func() { runDrainClusterBinary(hubClusterName, memberCluster1EastProdName, true) })
+
+	It("drain cluster using binary, should fail due to CRPDB", func() { runDrainClusterBinary(hubClusterName, memberCluster2EastCanaryName, false) })
+
+	It("should ensure no resources exist on drained clusters", func() {
+		for _, cluster := range drainClusters {
+			resourceRemovedActual := workNamespaceRemovedFromClusterActual(cluster)
+			Eventually(resourceRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to check if resources doesn't exist on member cluster")
+		}
+	})
+
+	It("should update cluster resource placement status as expected", func() {
+		crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), noDrainClusterNames, nil, "0")
+		Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update cluster resource placement status as expected")
+	})
+
+	It("should still place resources on the selected clusters which were not drained", func() {
+		for _, cluster := range noDrainClusters {
+			resourcePlacedActual := workNamespaceAndConfigMapPlacedOnClusterActual(cluster)
+			Eventually(resourcePlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place resources on the selected clusters")
+		}
+	})
+
+	It("uncordon cluster using binary", func() { runUncordonClusterBinary(hubClusterName, memberCluster1EastProdName) })
+
+	It("uncordon cluster using binary", func() { runUncordonClusterBinary(hubClusterName, memberCluster2EastCanaryName) })
+
+	It("should place resources on all available member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+})
+
+func runDrainClusterBinary(hubClusterName, memberClusterName string, isSuccess bool) {
+	By(fmt.Sprintf("draining cluster %s", memberClusterName))
+	cmd := exec.Command(drainBinaryPath,
+		"--hubClusterContext", hubClusterName,
+		"--clusterName", memberClusterName)
+	output, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred(), "Drain command failed with error: %v\nOutput: %s", err, string(output))
+	if isSuccess {
+		Expect(strings.Contains(string(output), "drain retry was successful for cluster")).To(BeTrue(), "Expected drain to be successful")
+	} else {
+		Expect(strings.Contains(string(output), "drain retry was not successful for cluster")).To(BeTrue(), "Expected drain to fail")
+	}
+}
+
+func runUncordonClusterBinary(hubClusterName, memberClusterName string) {
+	By(fmt.Sprintf("uncordoning cluster %s", memberClusterName))
+	cmd := exec.Command(uncordonBinaryPath,
+		"--hubClusterContext", hubClusterName,
+		"--clusterName", memberClusterName)
+	output, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred(), "Uncordon command failed with error: %v\nOutput: %s", err, string(output))
+	Expect(strings.Contains(string(output), "uncordoned member cluster")).To(BeTrue(), "Expected uncordon to be successful")
```

Comment:
Hi Arvind! IIRC the error check part already guarantees that the exit code must be 0; using the string match on logs to verify if it has run successfully is a bit error prone I am afraid.